src/game.cpp

```diff
@@ -1462,35 +1461,33 @@ bool Game::removeMoney(Cylinder* cylinder, uint64_t money, uint32_t flags /*= 0*
 			Container* tmpContainer = item->getContainer();
 			if (tmpContainer) {
 				containers.push_back(tmpContainer);
-			} else {
-				int32_t worth = item->getWorth();
-				if (worth != 0) {
-					moneyCount += worth;
-					moneyMap.emplace(worth, item);
-				}
+			} else if (item->getWorth() != 0) {
+				moneyCount += item->getWorth();
+				moneyMap.insert(moneymap_pair(item->getWorth(), item));
 			}
 		}
 	}
 
+	/*not enough money*/
 	if (moneyCount < money) {
 		return false;
 	}
 
-	for (const auto& moneyEntry : moneyMap) {
-		Item* item = moneyEntry.second;
-		if (moneyEntry.first >= money) {
-			uint32_t worth = moneyEntry.first / item->getItemCount();
-			uint32_t removeCount = (money / worth) + 1;
+	for (MoneyMap::const_iterator mit = moneyMap.begin(), mend = moneyMap.end(); mit != mend && money > 0; ++mit) {
```

Comment:
Why replace the range based for with the absolutely exact same, yet less semantic indexed for?